src/lib/dcob/octoclient.rb

```diff
@@ -28,6 +32,51 @@ def hookit(repository, callback_url)
       "Skipping #{repository} due to existing hook"
     end
 
+    def apply_commit_statuses(repository_id, pr_number)
```

Comment:
I moved this method into the Octoclient class, but with reservations. This is the switching logic on processing commit messages so it is pretty core to DCOB's business, not really a part of interacting with GitHub. It wound up this way because it was easier to spec this method here than any way I could think of so far to deal with the amount of setup required to do variations of commit collections with payloads through the actual handler.